hub/core/chunk.py

```diff
@@ -108,6 +114,33 @@ def update_headers(self, incoming_num_bytes: int, sample_shape: Tuple[int]):
         self.shapes_encoder.register_samples(sample_shape, 1)
         self.byte_positions_encoder.register_samples(num_bytes_per_sample, 1)
 
+    def update_sample(
+        self, local_sample_index: int, new_buffer: memoryview, new_shape: Tuple[int]
+    ):
+        # TODO: docstring
+
+        # TODO: warn user the length of buffer is longer than expected
```

Comment:
a better warning: warn the user when a completed chunk (one that will no longer be appended to) becomes less than a certain threshold or greater than a certain threshold.

for example, a user has chunks [16MB, 16MB, 18MB, 4MB], the last of which is unfinished.

if after an update, the new chunk sizes are [4MB, 16MB, 18MB, 4MB]

alter the user that the first chunk shrunk by 10MB